src/kv/test/kv_serialisation.cpp

```diff
@@ -291,6 +382,137 @@ TEST_CASE(
   }
 }
 
+struct CustomClass2
```

Comment:
Here's the real purpose of this PR:

`CustomClass2` has far fewer requirements than `CustomClass`. You don't need to `DECLARE_JSON_TYPE` and `MSGPACK_DEFINE` (you needed both in case `USE_NLJSON_SERIALISER` got turned on), it doesn't need to be hashable (just less-comparable). You may _want_ to define one of these things so you can use our standard serialisers, but you can also plug in your own.